src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,119 @@
-export const Pagination = () => {};
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pageNumbers = useMemo(() => {
+    const numbers = [];
+
+    for (let i = 1; i <= Math.ceil(total / perPage); i += 1) {
+      numbers.push(i);
+    }
+
+    return numbers;
+  }, [total, perPage]);
+
+  const handlePrevClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
+    const page = currentPage - 1;
+
+    event.preventDefault();
+
+    if (page > 0) {
+      onPageChange(page);
+    }
+  };
+
+  const handleNextClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
+    const page = currentPage + 1;
+
+    event.preventDefault();
+
+    if (page <= pageNumbers.length) {
+      onPageChange(page);
+    }
+  };
+
+  const handleNumberClick = (
+    event:
+    React.MouseEvent<HTMLAnchorElement>,
+    pageNumber: number,
+  ) => {
+    event.preventDefault();
+    onPageChange(pageNumber);
+  };
+
+  return (
+    <ul className="pagination">
+      <li
+        className={classNames(
+          'page-item',
+          {
+            disabled: currentPage === 1,
+          },
+        )}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
```

Comment:
let's update it.
In this case, you added "#prev" href for this tag a, and to disable reload page you added event.preventDefault() in handlePrevClick.

You can make it easier, by just using href="javascript:void(0)".
With 'javascript:void(0)' page will not be reloaded, and you don't need to use preventDefault().